src/WrappedMToken.sol

```diff
@@ -526,7 +526,8 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
         unchecked {
             uint112 principal_ = IndexingMath.getPrincipalAmountRoundedDown(amount_, currentIndex_);
 
-            if (principal_ >= _principalOfTotalEarningSupply || amount_ >= totalEarningSupply) {
+            // NOTE: Do not zero out `totalEarningSupply` just because `_principalOfTotalEarningSupply` is or will be 0.
```

Comment:
why ? when this use case of non-zeroing out is valid?

Reply:
`principal_` is a rounded down division from `amount_`, and in order to keep the inequality of `totalEarningSupply > sum of all earning balances`, then `_principalOfTotalEarningSupply` being reduced to 0 cannot imply that `sum of all earning balances` is 0.